### PR TITLE
Add Codon Table JSON input / output

### PR DESCRIPTION
**Pull request by Koeng101**
This adds features+test to import or export codon tables as JSON files. This does NOT add in that feature for the command line interface - I am building it to import as part of a webservice that supplies codon tables as JSON. 

Probably good to add it into the CLI, but honestly I might need some instruction on how to do that. I can set up an issue to do that, probably can come later.

### CODE REVIEW

File: transformations_test.go [hunk 1]
```diff
@@ -103,3 +107,50 @@ func TestGetCodonFrequency(t *testing.T) {
...
 }
+
+/******************************************************************************
+
+JSON related tests begin here.
+
+******************************************************************************/
+
+func ExampleReadCodonJSON() {
+	codontable := ReadCodonJSON("data/codontables/bsub_test.json")
```
[the remark this answers is not in the document]
Comment (TimothyStiles) [reply transformations_test.go:118]: Also no subdirectories in data directory.

**Review by TimothyStiles — DISMISSED** [review]

File: transformations.go [hunk 2]
```diff
@@ -359,3 +355,115 @@ var defaultCodonTablesByName = map[string]CodonTable{
...
+Also, I need codon tables in JSON for the codon optimizer app. That is
+the real reason I'm doing this now.
+
+Jolly Good!
+Keoni
+
+PS: On it @Tim, publishing those JSON representations.
+******************************************************************************/
+
+// ParseCodonJSON parses a CodonTable JSON file and adds appropriate pointers to struct.
```
[the remark this answers is not in the document]
Comment (Koeng101) [reply transformations.go:448]:
```go
// ParseJSON parses a Sequence JSON file and adds appropriate pointers to struct.
func ParseJSON(file []byte) Sequence {
        var sequence Sequence
        json.Unmarshal([]byte(file), &sequence)
        legacyFeatures := sequence.Features
        sequence.Features = []Feature{}

        for _, feature := range legacyFeatures {
                sequence.AddFeature(feature)
        }
        return sequence
}
```
How come this one does add appropriate pointers to struct? Are the pointers all the feature stuff? I'll remove that description, just want to know the reason (ie, naively I don't see any pointers in that code)

**Review by TimothyStiles — APPROVED** [review]
I approve these changes but perhaps there's someone else who can review it too?

File: data/bsub_codon_test.json [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+  "GTG"
+ ],
+ "stop_codons": [
+  "TAA",
+  "TAG",
+  "TGA"
+ ],
+ "amino_acids": [
+  {
+   "letter": "",
```
[the remark this answers is not in the document]
Comment (Koeng101) [reply data/bsub_codon_test.json]: Outdated

**Comment by Koeng101** [comment]
Any suggestions?

**Comment by TimothyStiles** [comment]
I'm not sure. What I think I'll do before we merge is actually look up what this PRs test coverage is then review it again or merge it if it's good enough.

**Comment by Koeng101** [comment]
Any way to have a little bot that tells us that on the PRs directly or something?

**Comment by Koeng101** [comment]
According to `go test cover`, the only line of code that is missing a test is 
```go
func ReadCodonJSON(path string) CodonTable {
        file, err := ioutil.ReadFile(path)
        if err != nil {
                // return 0, fmt.Errorf("Failed to open file %s for unpack: %s", gzFilePath, err)
        }
```

Not really sure how to test that, but basically I have coverage for things other than that.

**Pull request merged**